### PR TITLE
BaSP-TG-132-superadmin-redux

### PR DESCRIPTION
**Pull request by dayanagm96**
Implementing Redux on SuperAdmins for:
- Creation of constants.
- Creation of actions.
- Creation of reducers.
- Creation of thunks.

Implementing share components as well.

### CODE REVIEW

File: src/Components/SuperAdmins/SuperAdminFormEdit.jsx [hunk 1]
```diff
@@ -16,31 +22,109 @@ const EditSuperAdmin = (props) => {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [active, setActive] = useState();
+  const [showWarning1, setShowWarning1] = useState(false);
+  const [showWarning2, setShowWarning2] = useState(false);
+  const [showWarning3, setShowWarning3] = useState(false);
+  const [showWarning4, setShowWarning4] = useState(false);
+
+  const handleInput1 = (e) => {
```
Comment (Lucas-DeLorenzi) [inline src/Components/SuperAdmins/SuperAdminFormEdit.jsx:30]: the same suggestions as in the add form

File: src/Components/SuperAdmins/SuperAdminFormEdit.jsx [hunk 2]
```diff
@@ -55,56 +139,68 @@ const EditSuperAdmin = (props) => {
...
-          <label>Active</label>
-          <input
-            type="text"
-            name="active"
-            value={active}
-            onChange={(event) => setActive(event.target.value)}
-          ></input>
+          <Dropdown name="active" labelText="Set if is active" onChange={handleChange}></Dropdown>
         </div>
         <Button
```
Comment (Lucas-DeLorenzi) [inline src/Components/SuperAdmins/SuperAdminFormEdit.jsx:205]: the same suggestions as in the add form, about modal and her buttons and actions

File: src/Components/redux/modules/superAdmins/thunks.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      console.error(error);
+    }
+  };
+};
+
+export const addSuperAdmin = (superAdmin) => {
+  return async (dispatch) => {
+    dispatch(addSuperAdminsPending());
+    try {
+      const response = await fetch(`https://coco-trackgenix-server.vercel.app/SuperAdmins`, {
```
Comment (Lucas-DeLorenzi) [inline src/Components/redux/modules/superAdmins/thunks.js:51]: depending on the fetch result you could configure the buttons and messages that are displayed in the modals

File: src/Components/SuperAdmins/SuperAdminForm.jsx [hunk 4]
```diff
@@ -99,11 +211,10 @@ const AddSuperAdmin = (props) => {
...
-        <div>
-          <p>Success!</p>
-          <p>The Super Admin was successfully created</p>
-        </div>
+        <h2>{resStatus ? 'Success!' : 'Warning!'}</h2>
+        <h3 className={styles.modalMsg}>
+          {resStatus ? responseMsg : `The superAdmin could not be created because ${responseMsg}`}
+        </h3>
         <div>
           <Button
```
[the remark this answers is not in the document]
Comment (Lucas-DeLorenzi) [reply src/Components/SuperAdmins/SuperAdminForm.jsx:219]: clicking this button if the fields have errors, the screen turns white

File: src/Components/SuperAdmins/SuperAdminFormEdit.jsx [hunk 5]
```diff
@@ -55,56 +139,68 @@ const EditSuperAdmin = (props) => {
       });
   }, []);
 
+  if (isLoading) {
+    return <Loading className={styles.loadText}></Loading>;
+  }
   return (
     <div className={styles.container}>
       <Logo />
       <h2>Form Edit</h2>
```
Comment (Lucas-DeLorenzi) [inline src/Components/SuperAdmins/SuperAdminFormEdit.jsx:148]: in edit function it seems that the server accepts any type of input in the inputs: special characters, empty strings... it never fails

File: src/Components/redux/modules/superAdmins/thunks.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        superadmin.active = superadmin.active ? 'true' : 'false';
+      });
+      dispatch(getSuperAdminsSuccess(resp.data));
+    } catch (error) {
+      dispatch(getSuperAdminsError(error.toString()));
+    }
+  };
+};
+
+export const deleteSuperAdmins = (_id) => {
```
Comment (Lucas-DeLorenzi) [inline src/Components/redux/modules/superAdmins/thunks.js:32]: the same suggestion as in add function

File: src/Components/redux/modules/superAdmins/thunks.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      });
+      const resp = await response.json();
+      dispatch(addSuperAdminsSuccess(resp.data));
+    } catch (error) {
+      dispatch(addSuperAdminsError(error.toString()));
+    }
+  };
+};
+
+export const editSuperAdmin = (superAdmin, id) => {
```
Comment (Lucas-DeLorenzi) [inline src/Components/redux/modules/superAdmins/thunks.js:72]: the same suggestion as in add function

**Review by Lucas-DeLorenzi — CHANGES_REQUESTED** [review]

**Review by Lucas-DeLorenzi — APPROVED** [review]

**Review by DeepFreq — APPROVED** [review]

**Review by Juanoauil — APPROVED** [review]

**Pull request merged**